src/components/Header.jsx

```diff
@@ -1,99 +1,198 @@
 import { Link } from 'react-router-dom';
+
 import '../styles/Header.css'
+
 import { useEffect, useState } from 'react';
 
+
+
 function Header() {
+
     const [isScrolled, setIsScrolled] = useState(false);
+
     const [activePage, setActivePage] = useState('/');
+
     const [isMenuOpen, setIsMenuOpen] = useState(false);
 
+
+
     useEffect(() => {
+
         const handleScroll = () => {
+
             const currentScrollY = window.scrollY
 
+
+
             if (currentScrollY > 50) {
+
                 setIsScrolled(true);
+
             } else {
+
                 setIsScrolled(false);
+
             }
+
         }
+
         window.addEventListener('scroll', handleScroll);
+
         return () => window.removeEventListener('scroll', handleScroll);
+
     }, []);
 
+
+
     const handleLinkClick = (page) => {
+
         setActivePage(page);
+
         setIsMenuOpen(false);
+
     };
 
+
+
     const toggleMenu = () => {
+
         setIsMenuOpen(!isMenuOpen);
+
     };
```

Comment:
Should be a callback, => do not want to recompile this every render
```suggestion
    const handleLinkClick = React.useCallback((page) => {
        setActivePage(page);
        setIsMenuOpen(false);
    }, []);



    const toggleMenu =  React.useCallback(() => {
        setIsMenuOpen(!isMenuOpen);
    }, []);
```

Reply:
Actually you need to toggle so the last one is either:

```
const toggleMenu =  React.useCallback(() => {
        setIsMenuOpen(!isMenuOpen);
    }, [isMenuOpen]);
    ```
    
    so it re renders onyl when isMenuOpen is updated or better yet:
    
    ```
const toggleMenu =  React.useCallback(() => {
        setIsMenuOpen(prev => !prev);
    }, []);
    ```